lxd/daemon.go

```diff
@@ -116,6 +117,11 @@ type Daemon struct {
 
 	ctx    context.Context
 	cancel context.CancelFunc
+
+	// Stores the time the metrics were last fetched. This will be used to prevent stressing the instances too much.
+	metricsLastFetchedTime time.Time
+	// Cached metrics which are returned instead of querying all instances.
+	metrics *metrics.MetricSet
```

Comment:
I think we need a Mutex around that to avoid concurrent access/changes which would cause a panic.